dojo/tools/intsights/parser.py

```diff
@@ -0,0 +1,101 @@
+import json
+
+from dojo.models import Finding
+
+
+class IntSightsParser(object):
+    """
+    IntSights Threat Intelligence Feed
+    """
+
+    def get_scan_types(self):
+        return ["IntSights Report"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "IntSights Report"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "IntSights report file can be imported in JSON format."
+
+    def _parse_assets(self, assets):
+        """
+        Parses the Assets node to create a string to be used in the Description
+        """
+
+        assets_affected = ''
+
+        for entry in assets:
+            assets_affected = f'{assets_affected} Type: {entry["Type"]}, Value: {entry["Value"]},'
+
+        return assets_affected[:-1]
+
+    def get_findings(self, file, test):
+        duplicates = dict()
+
+        if not file:
+            return []
+
+        data = file.read()
+
+        try:
+            findings = json.loads(str(data, 'utf-8'))
+        except:
+            findings = json.loads(data)
+
+        if not findings:
+            raise ValueError('IntSights report contains errors: No vulnerabilities were found in the data provided')
+
+        if findings.get('Findings'):
+            for finding in findings['Findings']:
+                unique_id_from_tool = finding['_id']
+                title = finding['Details']['Title']
+
+                assets_affected = self._parse_assets(finding['Assets'])
+
+                description = f'{finding["Details"]["Description"]}' \
+                              f'\r\n\r\n----' \
+                              f'\r\n**Date Found**: {finding.get("FoundDate", "None provided")}' \
+                              f'\r\n**Date Updated**: {finding.get("UpdateDate", "None provided")}' \
+                              f'\r\n\r\n----' \
+                              f'\r\n\r\n**Type**: {finding["Details"]["Type"]}' \
+                              f'\r\n**SubType**: {finding["Details"]["SubType"]}' \
+                              f'\r\n**Source**: {finding["Details"]["Source"].get("URL", "None provided")}' \
+                              f'\r\n**Source Date**: {finding["Details"]["Source"].get("Date", "None provided")}' \
+                              f'\r\n**Source Type**: {finding["Details"]["Source"].get("URL", "N/A")}' \
+                              f'\r\n**Source Network Type**: {finding["Details"]["Source"].get("NetworkType", "N/A")}' \
+                              f'\r\n\r\n----' \
+                              f'\r\n**Assets Affected**: {assets_affected}' \
+                              f'\r\n\r\n----' \
+                              f'\r\n**Takedown Status**: {finding["TakedownStatus"]}'
+                severity = finding['Details']['Severity']
+                mitigation = "N/A"
+                impact = "N/A"
```

Comment:
These attribute are not mandatory for parsers.
So I think you can remove these lines and the attributes in the Finding constructor

Reply:
boilerplate from cookie cutter!